Source/Extensions/NSURLQueryItem.swift

```diff
@@ -0,0 +1,41 @@
+public extension NSURLQueryItem {
+  static func initAndEncode(key: String, value: String) -> NSURLQueryItem? {
+    guard let encodedKey = encode(key),
+          let encodedValue = encode(value)
+      else { return .None }
+
+    return NSURLQueryItem(name: encodedKey, value: encodedValue)
+  }
+
+  static func decode(data: NSData?) -> [NSURLQueryItem] {
+    guard let data = data else { return [] }
+    let result = String(data: data, encoding: NSUTF8StringEncoding)
+      .flatMap { $0.componentsSeparatedByString("&") }?
+      .map { $0.componentsSeparatedByString("=") }
+      .reduce([], combine: stringToQueryItem)
```

Comment:
I think this could just be a map, no?
You're taking an array of strings and turning them into an array of NSURLQueryItems.